File: C++/791. Custom Sort String.cpp
class Solution {
public:
    string customSortString(string o, string s) {
        map<char,int>m;
        string a="",b="";
        for(char c:s){
            m[c]++;
        }
        for(char c:o){
            while(m[c]!=0){
                a+=c;
                m[c]--;
            }
        }
        for(char c:s){
            if(m[c]!=0)
            b+=c;
        }
        sort(b.begin(),b.end());
        a+=b;
        return a;
    }
};

File: JAVA/791. Custom Sort String.java
class Solution {
    public String customSortString(String o, String s) {
        int[] f=new int[26];
        String a="",b="";
        for(int i=0;i<s.length();i++){
            f[s.charAt(i)-'a']++;
        }
        for(int i=0;i<o.length();i++){
            while(f[o.charAt(i)-'a']!=0){
                a+=o.charAt(i);
                f[o.charAt(i)-'a']--;
            }
        }
        for(int i=0;i<s.length();i++){
            if(f[s.charAt(i)-'a']!=0)
            b+=s.charAt(i);
        }
        String x=new String(b);
        a+=b;
        return a;
    }
}